core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -996,12 +994,12 @@ public ServerBuilder annotatedService(String pathPrefix, Object service,
         requireNonNull(exceptionHandlerFunctions, "exceptionHandlerFunctions");
         requireNonNull(requestConverterFunctions, "requestConverterFunctions");
         requireNonNull(responseConverterFunctions, "responseConverterFunctions");
-
-        final List<AnnotatedHttpServiceElement> elements =
-                AnnotatedHttpServiceFactory.find(pathPrefix, service, exceptionHandlerFunctions,
-                                                 requestConverterFunctions, responseConverterFunctions);
-        registerHttpServiceElement(elements, decorator);
-        return this;
+        return annotatedService().pathPrefix(pathPrefix)
+                                 .decorator(decorator)
+                                 .exceptionHandlers(exceptionHandlerFunctions)
+                                 .requestConverters(requestConverterFunctions)
+                                 .responseConverters(responseConverterFunctions)
+                                 .build(service);
```

Comment:
It's not your fault, but would you mind updating the method signature of this method so it accepts `Iterable<? extends {ExceptionHandler,RequestConverter,ResponseConverter}Function>`?